File: Travel Package Management Java/src/com/example/travel/ActivityTest.java
package com.example.travel;

import org.junit.Test;
import static org.junit.Assert.*;

public class ActivityTest {

    @Test
    public void testConstructor() {
        Activity activity = new Activity("NightWalk", "Exploring the night life", 100, 15);
        assertEquals("NightWalk", activity.getName());
        assertEquals(100, activity.getCost(), 0.01); // The third parameter is delta for floating point comparisons
    }

    @Test
    public void testSignUp() {
        Activity activity = new Activity("Trekking", "Trekking the mountains", 50, 2);
        Passenger passenger1 = new StandardPassenger("Amaan", 1, 200);
        Passenger passenger2 = new StandardPassenger("Atif", 2, 200);
        assertTrue(activity.signUp(passenger1));
        assertTrue(activity.signUp(passenger2));
        Passenger passenger3 = new StandardPassenger("Aisha", 3, 200);
        assertFalse(activity.signUp(passenger3)); // Should be false since capacity is 2
    }
}

File: Travel Package management C++/ActivityTest.cpp
#include "Activity.h"
#include "StandardPassenger.h" 
#include <gtest/gtest.h>
#include <memory>

TEST(ActivityTest, Constructor) {
    Activity activity("NightWalk", "Exploring the night life", 100, 15);
    EXPECT_EQ(activity.getName(), "NightWalk");
    EXPECT_EQ(activity.getCost(), 100);
}

// Test the signUp function for capacity limits
TEST(ActivityTest, SignUp) {
    Activity activity("Trekking", "Trekking the mountains", 50, 2);
    auto passenger1 = std::make_shared<StandardPassenger>("Amaan", 1, 200); // Use StandardPassenger
    auto passenger2 = std::make_shared<StandardPassenger>("Atif", 2, 200); // Use StandardPassenger
    EXPECT_TRUE(activity.signUp(passenger1));
    EXPECT_TRUE(activity.signUp(passenger2));
    auto passenger3 = std::make_shared<StandardPassenger>("Aisha", 3, 200); // Use StandardPassenger
    EXPECT_FALSE(activity.signUp(passenger3)); // Should be false since capacity is 2
}
